applications/callflow/src/module/cf_dynamic_cid.erl

```diff
@@ -142,28 +145,24 @@ handle_manual(Data, Call) ->
 %%--------------------------------------------------------------------
 -spec handle_list(wh_json:object(), whapps_call:call()) -> 'ok'.
 handle_list(Data, Call) ->
-    CallerIdNumber = whapps_call:caller_id_number(Call),
-    lager:debug("callerid number before this module: ~s ", [CallerIdNumber]),
-
     {NewCidInfo, Dest} = get_list_entry(Data, Call),
-
     NewCallerIdNumber = wh_json:get_value(<<"number">>, NewCidInfo),
     NewCallerIdName = wh_json:get_value(<<"name">>, NewCidInfo),
+    proceed_with_call(NewCallerIdName, NewCallerIdNumber, Dest, Data, Call).
 
-    lager:info("setting the caller id number to ~s", [NewCallerIdNumber]),
+-spec handle_lists(wh_json:object(), whapps_call:call()) -> 'ok'.
+handle_lists(Data, Call) ->
+    [NewCallerIdName, NewCallerIdNumber, Dest] = get_lists_entry(Data, Call),
+    proceed_with_call(NewCallerIdName, NewCallerIdNumber, Dest, Data, Call).
 
+-spec proceed_with_call(ne_binary()|'undefined', ne_binary()|'undefined', binary(), wh_json:object(), whapps_call:call()) -> 'ok'.
+proceed_with_call(NewCallerIdName, NewCallerIdNumber, Dest, Data, Call) ->
     Updates = [{fun whapps_call:kvs_store/3, 'dynamic_cid', NewCallerIdNumber}
                ,{fun whapps_call:set_caller_id_number/2, NewCallerIdNumber}
                ,{fun whapps_call:set_caller_id_name/2, NewCallerIdName}
               ],
-
     cf_exe:set_call(whapps_call:exec(Updates, Call)),
-
-    lager:debug("destination number from cf_capture_group regex: ~s ", [Dest]),
-    Number = wnm_util:to_e164(Dest),
-    lager:info("send the call onto real destination of: ~s", [Number]),
-
-    maybe_route_to_callflow(Data, Call, Number).
```

Comment:
Why remove the log lines?
Without them and if `to_e164/1` fails, it will be harder to debug.